common/types/offer.go

```diff
@@ -31,19 +36,41 @@ func HexToHash(s string) (Hash, error) {
 
 // Offer represents a swap offer
 type Offer struct {
-	ID            Hash
+	id            Hash
 	Provides      ProvidesCoin
 	MinimumAmount float64
 	MaximumAmount float64
 	ExchangeRate  ExchangeRate
 }
 
+// NewOffer creates and returns an Offer with an initialised id field
+func NewOffer(coin ProvidesCoin, minAmount float64, maxAmount float64, exRate ExchangeRate) *Offer {
+	offer := &Offer{
+		Provides:      coin,
+		MinimumAmount: minAmount,
+		MaximumAmount: maxAmount,
+		ExchangeRate:  exRate,
+	}
+	offer.setID()
+	return offer
+}
+
 // GetID returns the ID of the offer
 func (o *Offer) GetID() Hash {
-	if o.ID != [32]byte{} {
-		return o.ID
+	if o.id.IsZero() {
+		panic("offer was improperly initialised")
+	}
+	return o.id
+}
+
+// setID returns the ID of the offer
+func (o *Offer) setID() {
+	if !o.id.IsZero() {
+		panic("attempt to set offer ID more than once")
 	}
 
+	// TODO: If we are not saving the random nonce below, so we can't recreate the ID, why do we want to
+	//       include the serialised data when generating the ID?
```

Comment:
what do you mean by "why do we want to include the serialised data when generating the ID?" 

i guess my original intention was to have the offer ID be based on the contents, but multiple users could make the same offer so i added the random nonce. but yeah, generating a random ID number would be just as effective

Reply:
Yes, I was wondering why we serialized any data. If the step is not adding any value, it would make sense to strip it out.